doc/cephfs/charmap.rst

```diff
@@ -0,0 +1,185 @@
+CephFS Directory Entry Name Normalization and Case Folding
+==========================================================
+
+CephFS allows configuring directory trees to **normalize** or **case fold**
+directory entry names. This is typically a useful property for file systems
+exported by gateways like Samba which enforce a case-insensitive view of the
+file system, typically with performance penalties on file systems which are not
+case-insensitive.
+
+The following virtual extended attributes control the **character mapping**
+rules for directory entries:
+
+* ``ceph.dir.casesensitive``: A boolean setting for the case sensitivity of the directory.
+* ``ceph.dir.normalization``: A string setting for the type of Unicode normalization to use. Currently the normalization forms D (``nfd``), C (``nfc``), KD (``nfkd``), and KC (``nfkc``) are understood by the client.
+* ``ceph.dir.encoding``: A string setting for the encoding to use and enforce for directory entry names. The default and presently only supported encoding is UTF-8 (``utf8``).
```

Comment:
Is `string` here superfluous?

Reply:
the distinction is probably not significant but it can't hurt to be specific